### PR TITLE
ECIP-1095:feat: add vanilla sha-3 ecip draft

### PR DESCRIPTION
**Pull request by stevanlohja**
#### View draft: https://github.com/ethereumclassic/ECIPs/blob/8790db36f475a5e29d1c3ea5a22b5e16c60251bf/_specs/ecip-vanilla-sha-3.md

### CODE REVIEW

File: _specs/ecip-vanilla-sha-3.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+```
+ECIP: TBD
+Title: Change the ETC Proof of Work Algorithm to _vanilla_ Sha-3
+Status: Draft
+Type: Network
```
Comment (q9f) [inline _specs/ecip-vanilla-sha-3.md:5]: Can you make it Standards Track / Core?

File: _specs/ecip-vanilla-sha-3.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+```
+ECIP: TBD
```
[the remark this answers is not in the document]
Comment (q9f) [reply _specs/ecip-vanilla-sha-3.md:2]: can you also rename it to `ecip-1095.md`?

**Review by q9f — CHANGES_REQUESTED** [review]

File: _specs/ecip-vanilla-sha-3.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevanlohja) [reply _specs/ecip-vanilla-sha-3.md]: Updated the file and header to be ecip-1095

File: _specs/ecip-vanilla-sha-3.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  - Get commitment from miners and pools to contribute to the new PoW network.
+  - Have community recounciliation of testing results and make changes as needed.
+- __Set a Hard-fork date__: Assuming the Sha-3 testnet is deemed suitable for mainnet adoption, plan a hard-fork with enough time to coodinate stakeholders.
+
+References:
+1. Alexander Tsankov. ECIP-1049 Change the ETC Proof of Work Algorithm to Keccak256. URL: https://ecips.ethereumclassic.org/ECIPs/ecip-1049
+2. Federal Information Processing Standards (FIPS). PUB 202. 2015. URL: https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf 
+3. NIST. Approved Algorithms. URL: https://csrc.nist.gov/projects/hash-functions
+
+
```
[the remark this answers is not in the document]
Comment (stevanlohja) [reply _specs/ecip-vanilla-sha-3.md]: Added license to footer

File: _specs/ecip-vanilla-sha-3.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+```
+ECIP: TBD
+Title: Change the ETC Proof of Work Algorithm to _vanilla_ Sha-3
+Status: Draft
+Type: Network
+Discussion: TBD
```
[the remark this answers is not in the document]
Comment (stevanlohja) [reply _specs/ecip-vanilla-sha-3.md]: Added the discussion url.

File: _specs/ecip-vanilla-sha-3.md (hunk 1 above)
Comment (stevanlohja) [reply _specs/ecip-vanilla-sha-3.md]: Update the status to Standards Track/ Core.

File: _specs/ecip-1095.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+```
+ECIP: ECIP-1095
```
Comment (q9f) [inline _specs/ecip-1095.md:2]:
```suggestion
ECIP: 1095
```

File: _specs/ecip-1095.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Title: Change the ETC Proof of Work Algorithm to _vanilla_ Sha-3
+Status: Draft
+Type: Standards Track / Core
```
Comment (q9f) [inline _specs/ecip-1095.md:5]:
```suggestion
type: Standards Track
category: Core
```

**Review by q9f — CHANGES_REQUESTED** [review]
can you please copy-paste the header from another ECIP? this will break the website otherwise.

e.g., https://raw.githubusercontent.com/ethereumclassic/ECIPs/master/_specs/ecip-1049.md

**Review by TheEnthusiasticAs — APPROVED** [review]

**Comment by stevanlohja** [comment]
> can you please copy-paste the header from another ECIP? this will break the website otherwise.
> 
> e.g., https://raw.githubusercontent.com/ethereumclassic/ECIPs/master/_specs/ecip-1049.md

My latest commit should now reflect these changes.

File: _specs/ecip-1095.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+# Change the ETC Proof of Work Algorithm to Sha-3-256 (vanilla Sha-3)
+
+__Abstract.__ Make Sha-3-256 Standard the Proof of Work algorithm for Ethereum Classic. Ethereum’s development coincided with the development of the Sha-3 standard, but the Sha-3 standard was finalized shortly after Ethereum launched. So, Ethereum was left with a variant of Sha-3 that did not produce standard Sha-3 hashes. This variant of Sha-3 utilized in Ethereum was named Keccak-256 to avoid confusion. Keccak-256 is utilized in the EVM, but the Proof of Work algorithm is Ethash. Ethash is a combination of Keccak256 & Daggerhashimoto which contributes to the memory-intensive feature (Dag). The Dag was added due to the premise of ASIC resistance, but that premise is flase since Ethash ASICs exist. So, Ethash’s Dag is built on the false premise of ASIC resistance and not utilizing the official Sha-3 Standard algorithm.
+
+Ethereum Classic's hash rate is also a minority of Ethereum's and the amount of rentable hash rate is available on the market to 51% attack the Ethereum Classic network. In summary, this proposal seeks to:
+
+- Remove Daggerhashimoto (Dag)
+- Bump Keccak-256 to Sha-3-256 Standard
```
Comment (q9f) [inline _specs/ecip-1095.md:20]: Wouldn't that break everything? I don't think we can _change_ that, all we can do is having SHA3 in addition to Keccak256.

File: _specs/ecip-1095.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+created: 2020-08-20
+---
+
+# Change the ETC Proof of Work Algorithm to Sha-3-256 (vanilla Sha-3)
+
+__Abstract.__ Make Sha-3-256 Standard the Proof of Work algorithm for Ethereum Classic. Ethereum’s development coincided with the development of the Sha-3 standard, but the Sha-3 standard was finalized shortly after Ethereum launched. So, Ethereum was left with a variant of Sha-3 that did not produce standard Sha-3 hashes. This variant of Sha-3 utilized in Ethereum was named Keccak-256 to avoid confusion. Keccak-256 is utilized in the EVM, but the Proof of Work algorithm is Ethash. Ethash is a combination of Keccak256 & Daggerhashimoto which contributes to the memory-intensive feature (Dag). The Dag was added due to the premise of ASIC resistance, but that premise is flase since Ethash ASICs exist. So, Ethash’s Dag is built on the false premise of ASIC resistance and not utilizing the official Sha-3 Standard algorithm.
+
+Ethereum Classic's hash rate is also a minority of Ethereum's and the amount of rentable hash rate is available on the market to 51% attack the Ethereum Classic network. In summary, this proposal seeks to:
+
+- Remove Daggerhashimoto (Dag)
```
Comment (q9f) [inline _specs/ecip-1095.md:19]: I don't think we can remove the DAG, we can just stop using it.

File: _specs/ecip-1095.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+- Bump Keccak-256 to Sha-3-256 Standard
+- Become a majority PoW algorithm in the industry
```
Comment (q9f) [inline _specs/ecip-1095.md:21]: Why not Keccak256?

File: _specs/ecip-1095.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+__Disclaimer.__ This is an alternative proposal to ECIP-1049 [1] since Keccak-256 and _vanilla_ Sha-3 produce different hashing results. However, most of the arguments in this proposal are compatible with ECIP-1049 since Keccak-256 and _vanilla_ Sha-3 differ in padding. ECIP-1049 and this proposal are both in support of switching the PoW to a Sha-3-based algorithm.
+
+__Motivation.__ The motivation of this proposal is to standardize the cryptographic hash function of the Ethereum Classic network with _vanilla_ Sha-3 and become a majority chain in it's respective PoW.
+
+__Rationale.__ 
+
+- __Cryptographic standardization__: By bumping Keccak-256 to _vanilla_ Sha-3, Ethereum Classic will have the official Sha-3 algorithm recognized by NIST. Sha-3 features:
+  - __Enhanced security__: Sha-3 is the latest member of the Secure Hash Algorithm family of standards and certified by the Federal Information Processing Standards (FIPS) [2].
```
Comment (q9f) [inline _specs/ecip-1095.md:30]: One could argue that not having the final standard actually adds sovereignty and arguably a layer of security.

File: _specs/ecip-1095.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+- __Less PoW Competition__: There are no major public blockchains whom use Sha-3 for PoW. Ethereum Classic's current market status to adopt Sha-3 would make it the majority chain for its respective PoW without having to compete in existing PoW ecosystems for miners.
+
+__Design.__
+
+As mentioned in ECIP-1049:
+
+> 1. The Proof of work function needs to be replaced with Keccak256
+> 2. The Function that checks the nonce-header in the block needs to know to accept Keccak256 hashes as valid for a block.
+
+Diagram: https://camo.githubusercontent.com/97974a2041c7633669309cf41902e4ad6b78e4cc/68747470733a2f2f692e696d6775722e636f6d2f32686f62714f4c2e706e67
```
Comment (q9f) [inline _specs/ecip-1095.md:42]: Can you add this to the assets and render it as diagram?

File: _specs/ecip-1095.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+Instead: 
+
+1. Replace the PoW function in the EVM with _vanilla_ Sha-3-256.
+2. The function that checks the nonce-header in a block needs to accept _vanilla_ Sha-3 as valid.
```
Comment (q9f) [inline _specs/ecip-1095.md:47]: How would a transition look like from Ethash to SHA-3?

File: _specs/ecip-1095.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+__Development & Testing.__
+
+Since switching to Sha-3 is resetting the mining algorithm, we need "open-source" tools and resources to help facilitate the adoption of the new PoW.
+
+- __Client implementation__: All major Ethereum Classic protocol providing clients such as Core-geth and Besu need a Sha-3 implementation.
+- __Mining-pool software__: A mining pool tool will allow anyone to contribute a mining pool to the Etherem Classic network.
+- __Mining software__: Existing AMD+NVIDIA GPU mining software for Ethash is available, but there is not solidified Sha-3 variant. We can collaborate with existing mining software providers or build new ones.
+- __Mining hardware__: We need to know the performance difference between commodity GPUs and a Sha-3 ASIC. Sha-3 is ASIC friendly and cheaper to build ASICs for than Ethash. Some chip manufacturers may be interested in participating in the initiative since is may be economically rewarding.
+- __Sha-3 testnet launch__: Launch of a Sha-3 Ethereum Classic testnet.
```
Comment (q9f) [inline _specs/ecip-1095.md:57]: It also needs a live testnet switch on Mordor.

File: _specs/ecip-1095.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+- __Sha-3 network testing__: Test all operations of the testnet from mining, transactions, and smart contracts.
+
+__Implementation.__
+
+- __Recounciliation__: 
+  - Get commitment from miners and pools to contribute to the new PoW network.
+  - Have community recounciliation of testing results and make changes as needed.
+- __Set a Hard-fork date__: Assuming the Sha-3 testnet is deemed suitable for mainnet adoption, plan a hard-fork with enough time to coodinate stakeholders.
```
Comment (q9f) [inline _specs/ecip-1095.md:65]: I strongly believe we would need a hybrid transition period here. Having one block to change the algorithm will be madness.

**Review by q9f — APPROVED** [review]

**Comment by q9f** [comment]
Looking good from an editor perspective. It needs some work on the content. I'll repost in the discussion.

**Comment by block-head** [comment]
See my comments here - https://github.com/ethereumclassic/ECIPs/issues/13 - This ECIP should make it clear that only Ethash is being replaced, and that the difficulty calculation remains a contamination of the mixHash and nonce, where the hash result is converted to a 256 integer and compared against the difficulty values.  Otherwise it should be rejected on technical grounds as it would impose an unacceptable risk because of the engineering teams implementing the changes.

**Comment by block-head** [comment]
Second objection: is this also implying SHA3 as a blanket replacement for all uses of Keccak-256?  This would be _devastating_ to the ecosystem tooling.  All of the EthereumMainnet tooling would have to be forked because Keccak is used almost everywhere.  From transaction hashing to block identification hashing.  Client side and server side code would all have to be forked and variant copies maintained everywhere.

This broad of a change is like shooting yourself in the foot with a nuclear warhead.  This ECIP should have it's scope reduced to just hashing or summarily rejected for the sake of the ecosystem.

**Pull request merged**